runtime/src/bank.rs

```diff
@@ -1335,6 +1335,32 @@ impl Bank {
         (examined_count, rewritten_count)
     }
 
+    // Calculates the starting-slot for inflation from the activation slot.
+    // This method assumes that `pico_inflation` will be enabled before `full_inflation`, giving
+    // precedence to the latter. However, since `pico_inflation` is fixed-rate Inflation, should
+    // `pico_inflation` be enabled 2nd, the incorrect start slot provided here should have no
+    // effect on the inflation calculation.
+    fn get_inflation_start_slot(&self) -> Slot {
+        self.feature_set
+            .activated_slot(&feature_set::full_inflation::id())
+            .unwrap_or_else(|| {
+                self.feature_set
+                    .activated_slot(&feature_set::pico_inflation::id())
+                    .unwrap_or(0)
+            })
+    }
+
+    fn get_inflation_num_slots(&self) -> u64 {
+        let inflation_activation_slot = self.get_inflation_start_slot();
+        // Normalize inflation_start to align with the start of rewards accrual.
+        let inflation_start_slot = self.epoch_schedule.get_first_slot_in_epoch(
+            self.epoch_schedule
+                .get_epoch(inflation_activation_slot)
+                .saturating_sub(1),
+        );
+        self.epoch_schedule.get_first_slot_in_epoch(self.epoch()) - inflation_start_slot
```

Comment:
yeah, I like the new definition over the previous one (`self.epoch_schedule.get_last_slot_in_epoch(prev_epoch)`).